_health-care/_js/components/HealthCareApp.jsx

```diff
@@ -213,8 +213,30 @@ class HealthCareApp extends React.Component {
     this.setState({ applicationData: newApplicationData });
   }
 
+  resetNullValues(objectData) {
+    const context = this;
+    _.forIn(objectData, (value, key) => {
+      if (value === null) {
+        _.set(objectData, key, '');
+      } else if (typeof (value) === 'object') {
+        context.resetNullValues(value);
+      }
+    });
+
+    return objectData;
+  }
+
   handleContinue() {
-    hashHistory.push(this.getNextUrl());
+    const sectionPath = this.props.location.pathname.split('/').filter((path) => { return !!path; });
+    const sectionData = _.get(this.state.applicationData, sectionPath);
+
+    this.resetNullValues(sectionData);
+
+    this.setState({}, () => {
+      if (validationFunctions.isValidSection(sectionPath, sectionData)) {
+        hashHistory.push(this.getNextUrl());
+      }
```

Comment:
If not, then it just does nothing for now...should we put a TODO here to add a flash or something?  Or is it good enough UI notification to have all the boxes go red?